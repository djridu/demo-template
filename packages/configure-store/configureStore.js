/**
 * Create the store with dynamic reducers
 */

import { createStore, applyMiddleware, combineReducers, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';

const sagaMiddleware = createSagaMiddleware();

export default function configureStore(initialState = {}, reducers = {}, middlewares = []) {
    const middlewaresToApply = [sagaMiddleware, ...middlewares];
    const enhancers = [applyMiddleware(...middlewaresToApply)];

    const composeEnhancers =
        '__DEV__' && typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
            ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
            : compose;

    const store = createStore(
        combineReducers({ ...reducers }),
        initialState,
        composeEnhancers(...enhancers),
    );

    store.runSaga = sagaMiddleware.run;
    store.injectedReducers = reducers;
    store.injectedSagas = {};

    if (module.hot) {
        module.hot.accept(() => {
            store.replaceReducer(combineReducers({ ...store.injectedReducers }));
        });
    }

    return store;
}
