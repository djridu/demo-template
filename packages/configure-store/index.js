// @flow

export { default as configureStore } from './configureStore';
