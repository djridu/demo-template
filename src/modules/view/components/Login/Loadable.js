/**
 *
 * Asynchronously loads the component for Login
 *
 */

import loadable from 'lazy-loadable';

export default loadable(() => import('./index'));
