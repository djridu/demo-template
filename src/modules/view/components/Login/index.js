/**
 *
 * Login
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { Grid, Cell } from 'styled-css-grid';
// import { FormattedMessage } from 'react-intl';
// import messages from './messages';

const Container = styled.div`
    background-image: linear-gradient(241deg, #3c276b, #120f3d 98%);
`;

const LoginWrapper = styled.div`
    margin: auto;
    width: 420px;
    height: 474px;
    border-radius: 9px;
    box-shadow: 0 12px 48px 0 rgba(0, 0, 0, 0.25);
    background-color: #ffffff;
`;

const LoginHeader = styled.header`
    padding-top: 21px;
    padding-bottom: 21px;
    margin-bottom: 30px;
    box-shadow: 0 1px 0 0 #f3f2f7;
`;

const LoginHeading = styled.p`
    margin: 0;
    font-size: 30px;
    font-weight: bold;
    font-style: normal;
    font-stretch: normal;
    line-height: 1.6;
    letter-spacing: 0.5px;
    text-align: center;
    color: #3c276b;
`;

const LoginBody = styled.div`
    padding-right: 32px;
    padding-left: 32px;
`;

const LoginFieldset = styled.fieldset`
    position: relative;
    padding: 0;
    margin: 0;
    border: 0;

    & + & {
        margin-top: 24px;
    }

    &:nth-last-of-type(2) {
        margin-top: 32px;
    }

    &:last-of-type {
        text-align: center;
    }
`;

const LoginInput = styled.input`
    padding: 0 18px;
    width: 360px;
    height: 48px;
    border-radius: 6px;
    border: none;
    background-color: #f3f2f7;
    font-size: 12px;
    font-weight: 500;
    font-style: normal;
    font-stretch: normal;
    line-height: 2;
    letter-spacing: 0.5px;
    text-align: left;
    color: #b2b1bd;
`;

const LoginOptionsNote = styled.p`
    padding-top: 8px;
    display: block;
    width: 100%;
    font-size: 12px;
    text-align: center;
    text-transform: uppercase;
`;

const LoginButton = styled(Link)`
    display: block;
    width: 168px;
    height: 48px;
    border-radius: 6px;
    background-color: #ff9900;
    font-size: 14px;
    font-weight: 600;
    font-style: normal;
    font-stretch: normal;
    text-decoration: none;
    line-height: 45px;
    letter-spacing: 0.5px;
    text-align: center;
    color: #ffffff;
    box-shadow: 0 10px 10px rgba(0, 0, 0, 0.08);
    transition: all 0.25s cubic-bezier(0.02, 0.01, 0.47, 1);

    &:hover {
        box-shadow: 0 15px 15px rgba(0, 0, 0, 0.16);
        transform: translate(0, -5px);
    }

    &:active {
        background: #ff7b00;
        color: #fff;
        box-shadow: 0 10px 10px rgba(0, 0, 0, 0.08);
    }
`;

class Login extends React.Component {
    render() {
        return (
            <Container>
                <Grid columns="1fr" rows="120px 1fr 120px" areas={['header', 'content', 'footer']}>
                    <Cell area="header" center middle>
                        <h1>Header</h1>
                    </Cell>
                    <Cell area="content" center middle>
                        <LoginWrapper>
                            <LoginHeader>
                                <LoginHeading>Sign In</LoginHeading>
                            </LoginHeader>

                            <LoginBody>
                                <LoginFieldset>
                                    <LoginInput placeholder="E-mail" type="text" required />
                                </LoginFieldset>

                                <LoginFieldset>
                                    <LoginInput placeholder="Password" type="password" required />
                                </LoginFieldset>

                                <LoginFieldset>
                                    <LoginButton to="/home">Login</LoginButton>
                                </LoginFieldset>

                                <LoginFieldset>
                                    <LoginOptionsNote>Don't have account?</LoginOptionsNote>
                                </LoginFieldset>
                            </LoginBody>
                        </LoginWrapper>
                    </Cell>
                    <Cell area="footer" center middle>
                        <p>Footer</p>
                    </Cell>
                </Grid>
            </Container>
        );
    }
}

Login.propTypes = {};

export default Login;
