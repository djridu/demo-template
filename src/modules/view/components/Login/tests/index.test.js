/**
 *
 * Tests for Login
 *
 *
 */

import * as React from 'react';
import { cleanup, render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { configureStore } from 'configure-store';
import { Provider, ReactReduxContext } from 'react-redux';
import { ConnectedRouter, connectRouter } from 'connected-react-router';
import history from 'utils/history';
import languageProviderReducer from 'containers/LanguageProvider/reducer';
import Login from '../index';

describe('<Login />', () => {
    let store;

    beforeAll(() => {
        const reducers = {
            language: languageProviderReducer,
            router: connectRouter(history),
        };
        store = configureStore({}, reducers, []);
    });

    afterEach(cleanup);

    it('Expect to not log errors in console', () => {
        const spy = jest.spyOn(global.console, 'error');
        render(
            <Provider store={store} context={ReactReduxContext}>
                <IntlProvider locale="en">
                    <ConnectedRouter history={history}>
                        <Login />
                    </ConnectedRouter>
                </IntlProvider>
            </Provider>,
        );
        expect(spy).not.toHaveBeenCalled();
    });

    it('Should render and match the snapshot', () => {
        const { container } = render(
            <Provider store={store} context={ReactReduxContext}>
                <IntlProvider locale="en">
                    <ConnectedRouter history={history}>
                        <Login />
                    </ConnectedRouter>
                </IntlProvider>
            </Provider>,
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
