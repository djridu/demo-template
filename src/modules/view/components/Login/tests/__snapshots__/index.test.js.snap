// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Should render and match the snapshot 1`] = `
<div
  class="Container-sc-7dyfqj-0 dFxklx"
>
  <div
    class="Grid__Grid-bojwg dVngpS"
  >
    <div
      class="Cell__Cell-jsYrZv cOetvE"
    >
      <h1>
        Header
      </h1>
    </div>
    <div
      class="Cell__Cell-jsYrZv TpYWU"
    >
      <div
        class="LoginWrapper-sc-7dyfqj-1 daTQUk"
      >
        <header
          class="LoginHeader-sc-7dyfqj-2 fDmaJX"
        >
          <p
            class="LoginHeading-sc-7dyfqj-3 cTgkyy"
          >
            Sign In
          </p>
        </header>
        <div
          class="LoginBody-sc-7dyfqj-4 fYFufT"
        >
          <fieldset
            class="LoginFieldset-sc-7dyfqj-5 cRhLmR"
          >
            <input
              class="LoginInput-sc-7dyfqj-6 dyoVMa"
              placeholder="E-mail"
              required=""
              type="text"
            />
          </fieldset>
          <fieldset
            class="LoginFieldset-sc-7dyfqj-5 cRhLmR"
          >
            <input
              class="LoginInput-sc-7dyfqj-6 dyoVMa"
              placeholder="Password"
              required=""
              type="password"
            />
          </fieldset>
          <fieldset
            class="LoginFieldset-sc-7dyfqj-5 cRhLmR"
          >
            <a
              class="LoginButton-sc-7dyfqj-8 hDwIps"
              href="/home"
            >
              Login
            </a>
          </fieldset>
          <fieldset
            class="LoginFieldset-sc-7dyfqj-5 cRhLmR"
          >
            <p
              class="LoginOptionsNote-sc-7dyfqj-7 iowyjN"
            >
              Don't have account?
            </p>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="Cell__Cell-jsYrZv hGsICH"
    >
      <p>
        Footer
      </p>
    </div>
  </div>
</div>
`;
